examples/main_refactor.py

```diff
@@ -0,0 +1,165 @@
+
+    if not status:
+        warnings.warn("Simulation failed!")
+    else:
+        tavg_data = tavg_data[:, :, :, 0]
+
+    vois = ['x1', 'y1', 'z', 'x2', 'y2', 'g', 'lfp']
+    # vois custom
+    # vois = ['x1', 'z', 'x2']
+
+    sensorsSEEG = []
+    projections = []
+    for sensors, projection in head.sensorsSEEG.iteritems():
+        if projection is None:
+            continue
+        else:
+            projection = calculate_projection(sensors, head.connectivity)
+            head.sensorsSEEG[sensors] = projection
+            print projection.shape
+            sensorsSEEG.append(sensors)
+            projections.append(projection)
+
+    hpf_flag = False
+    hpf_low = max(16.0, 1000.0 / time_length)  # msec
+    hpf_high = min(250.0, fsAVG)
+
+    if status:
+
+        model = simulator_instance.model
+
+        logger.info("\nSimulated signal return shape: " + str(tavg_data.shape))
+        logger.info("Time: " + str(scale_time * ttavg[0]) + " - " + str(scale_time * ttavg[-1]))
+        logger.info("Values: " + str(tavg_data.min()) + " - " + str(tavg_data.max()))
+
+        write_h5_model(simulator_instance.prepare_for_h5(), folder_name=data_folder, file_name=hypothesis.get_name() + "_sim_settings.h5")
+
+        # Pack results into a dictionary, high pass filter, and compute SEEG
+        res = dict()
+        time = scale_time * numpy.array(ttavg, dtype='float32')
+        dt = numpy.min(numpy.diff(time))
+        for iv in range(len(vois)):
+            res[vois[iv]] = numpy.array(tavg_data[:, iv, :], dtype='float32')
+
+        if model._ui_name == "EpileptorDP2D":
+            raw_data = numpy.dstack([res["x1"], res["z"], res["x1"]])
+            lfp_data = res["x1"]
+            for i in range(len(projections)):
+                res['seeg' + str(i)] = numpy.dot(res['z'], projections[i].T)
+        else:
+            if model._ui_name == "CustomEpileptor":
+                raw_data = numpy.dstack([res["x1"], res["z"], res["x2"]])
+                lfp_data = res["x2"] - res["x1"]
+            else:
+                raw_data = numpy.dstack([res["x1"], res["z"], res["x2"]])
+                lfp_data = res["lfp"]
+            for i in range(len(projections)):
+                res['seeg' + str(i)] = numpy.dot(res['lfp'], projections[i].T)
+                if hpf_flag:
+                    for i in range(res['seeg'].shape[0]):
+                        res['seeg_hpf' + str(i)][:, i] = filter_data(res['seeg' + str(i)][:, i], hpf_low, hpf_high,
+                                                                     fsAVG)
+
+        write_ts_epi(raw_data, dt, lfp_data, path=os.path.join(FOLDER_RES, hypothesis.name + "_ep_ts.h5"))
+        del raw_data, lfp_data
+
+        for i in range(len(projections)):
+            write_ts_seeg_epi(res['seeg' + str(i)], dt, path=os.path.join(FOLDER_RES, hypothesis.name + "_ep_ts.h5"))
+
+        res['time'] = time
+
+        del ttavg, tavg_data
```

Comment:
restructure this main clause as a set of small functions and you won't need this.  you should never need `del` except at a module global scope.